specs/code-checks.md

```diff
@@ -0,0 +1,413 @@
+- Proper error handling
+
+**Style (enforced by formatter, not linter):**
+- Indent, spacing, line length, etc. should be handled by Prettier, not ESLint
+
+### Auto-Fixable Rules
+
+Where possible, ESLint rules SHOULD be configured with auto-fix capability. Developers SHOULD run auto-fix before committing:
+
+```bash
+pnpm run lint:fix  # Auto-fix all fixable issues
+```
+
+---
+
+## Formatting
+
+### Prettier Configuration
+
+The project MUST use Prettier for consistent code formatting. Prettier MUST be configured to:
+
+- Format TypeScript files (`.ts`)
+- Format JSON files (`.json`)
+- Format Markdown files (`.md`)
+- Use consistent style across the entire codebase
+
+### Formatting Requirements
+
+Code MUST be formatted according to Prettier rules before merge. Formatting MUST be enforced:
+
+- Via pre-commit hooks (auto-format staged files)
+- Via CI checks (verify formatting, fail if not formatted)
+- Via IDE integration (format on save recommended)
+
+**Formatting violations MUST fail CI** unless auto-formatting is enabled in pre-commit hooks.
+
+### Formatting Standards
+
+The Prettier configuration SHOULD use these settings as a baseline:
+
+- **Print width:** 100 characters (or 120 for longer lines)
+- **Tab width:** 2 spaces
+- **Semicolons:** Required
+- **Quotes:** Single quotes for strings, double quotes for JSX
+- **Trailing commas:** ES5-style (where valid in ES5)
+- **Arrow function parens:** Always include parentheses
+
+**[DEFERRED]** Exact Prettier configuration values are implementation decisions. The requirement is consistent formatting, not specific style choices.
+
+### Integration with ESLint
+
+Prettier and ESLint MUST NOT conflict. The ESLint configuration MUST:
+
+- Disable all style-related rules that conflict with Prettier
+- Use `eslint-config-prettier` to prevent conflicts
+- Run Prettier as a separate check (not through ESLint plugin)
+
+---
+
+## Pre-Commit Hooks
+
+### Hook Requirements
+
+The project SHOULD use pre-commit hooks to enforce code quality before commits reach the repository. Pre-commit hooks SHOULD run:
+
+- Type checking on staged files
+- Linting on staged files (with auto-fix)
+- Formatting on staged files (with auto-format)
+
+**Performance requirement:** Pre-commit hooks MUST complete in under 10 seconds for typical commits. If checks are too slow, consider only running on changed files or using incremental checking.
+
+### Hook Failures
+
+If pre-commit hooks fail:
+- Type errors MUST block the commit
+- Lint errors MUST block the commit (unless auto-fixed)
+- Format errors SHOULD be auto-fixed (block only if auto-fix fails)
+
+Developers MAY bypass hooks with `--no-verify` flag for emergency situations, but bypassed commits MUST still pass CI checks before merge.
+
+### Recommended Tools
+
+**[DEFERRED]** The choice of pre-commit hook implementation is an implementation decision. Options include:
+- `husky` + `lint-staged` (popular JavaScript ecosystem choice)
```

Comment:
I think it confused things to use these options here so let's standardise on husky and lint-staged.